client/state/data-layer/wpcom/i18n/locale-suggestions/index.js

```diff
@@ -0,0 +1,52 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { noop } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequestEx } from 'state/data-layer/wpcom-http/utils';
+import { I18N_LOCALE_SUGGESTIONS_REQUEST } from 'state/action-types';
+import { receiveLocaleSuggestions } from 'state/i18n/locale-suggestions/actions';
+
+/**
+ * @module state/data-layer/wpcom/i18n/locale-suggestions
+ */
+
+/**
+ * Dispatches a request to fetch locale suggestions
+ *
+ * @param {Object} action Redux action
+ * @returns {Object} original action
+ */
+export const fetchLocaleSuggestions = action =>
+	http(
+		{
+			apiVersion: '1.1',
+			method: 'GET',
+			path: '/locale-guess',
```

Comment:
Naming structure in the data-layer mirrors the wpcom api path, mainly to make lookup easier. So in this case the file here should be `client/state/data-layer/wpcom/locale-guess/index.js`

https://github.com/Automattic/wp-calypso/tree/master/client/state/data-layer#file-structure